File: bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/guide/VignettingGuideProbe.scala
package edu.gemini.spModel.guide

import edu.gemini.spModel.core.Coordinates
import edu.gemini.spModel.obs.context.ObsContext


trait VignettingGuideProbe {
  def calculateVignetting(ctx: ObsContext, guideStarCoordinates: Coordinates): Double
}

File: bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/guide/VignettingGuideProbe.java
package edu.gemini.spModel.guide;

import edu.gemini.spModel.obs.context.ObsContext;

/**
 * A guide probe that can vignette the science area.
 */
public interface VignettingGuideProbe extends GuideProbe {

    /**
     * Constructs a VignettingCalculator for this GuideProbe in the given
     * context.
     */
    VignettingCalculator calculator(ObsContext ctx);
}
